org-rainbow-tags.el

```diff
@@ -181,10 +205,12 @@ colors and want to change them. Should be between 0-100."
       (progn
         (org-rainbow-tags--apply-overlays)
         (add-hook 'org-after-tags-change-hook
-                  'org-rainbow-tags--apply-overlays))
+                  'org-rainbow-tags--apply-overlays)
+        (advice-add 'org-clock-report :after 'org-rainbow-tags--apply-overlays-advice))
     (org-rainbow-tags--delete-overlays)
     (remove-hook 'org-after-tags-change-hook
-                 'org-rainbow-tags--apply-overlays)))
+                 'org-rainbow-tags--apply-overlays)
+    (advice-remove 'org-clock-report 'org-rainbow-tags--apply-overlays-advice)))
```

Comment:
I tested this and `org-clock-report` doesn't evaluated with `C-c C-c` in the table. We can use `org-update-dblock`, it works both with `C-c C-c` and `M-x org-clock-report`.

Also, we don't need to create another function just for the advice, this works fine:
```elisp
(advice-add 'org-update-dblock :after (lambda (&rest _) (org-rainbow-tags--apply-overlays)))
(advice-remove 'org-update-dblock (lambda (&rest _) (org-rainbow-tags--apply-overlays)))
```

Reply:
A comment saying this advices are about `org clock table` will be nice though, we can forgot about it easily.

Reply:
We can merge this PR after this improvement is made, thank you.

Reply:
Done.

Reply:
I also added a comment for the advice now. I was working off a stale browser tab before, and it didn't show all your comments.